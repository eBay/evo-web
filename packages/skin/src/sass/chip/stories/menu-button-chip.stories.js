export default { title: "Skin/Chip/Menu Button Chip" };

export const collapsed = () => `
<button class="chip chip--filter" aria-expanded="false">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-chevron-down-12"/>
    </svg>
</button>
`;

export const expanded = () => `
<button class="chip chip--filter" aria-expanded="true">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-chevron-down-12"/>
    </svg>
</button>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip chip--discovery" aria-expanded="false">
        <span class="chip__text">Football</span>
        <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
            <use href="#icon-chevron-down-12"/>
        </svg>
    </button>
</div>
`;
