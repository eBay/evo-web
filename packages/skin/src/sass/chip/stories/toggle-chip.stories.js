export default { title: "Skin/Chip/Toggle Chip" };

export const button = () => `
<button class="chip" type="button" aria-pressed="false">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</button>
`;

export const buttonSelected = () => `
<button class="chip" type="button" aria-pressed="true">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</button>
`;

export const anchor = () => `
<a class="chip" herf="https://www.ebay.com">
    <span class="chip__text">Football</span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</a>
`;

export const anchorSelected = () => `
<a class="chip chip--selected" herf="https://www.ebay.com">
    <span class="chip__text">Football <span class="clipped">- Selected</span></span>
    <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
        <use href="#icon-close-12"></use>
    </svg>
</a>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip" type="button" aria-pressed="false">
        <span class="chip__text">Football</span>
        <svg class="icon icon--12 chip__trailing" width="13" height="12" aria-hidden="true">
            <use href="#icon-close-12"></use>
        </svg>
    </button>
</div>
`;
