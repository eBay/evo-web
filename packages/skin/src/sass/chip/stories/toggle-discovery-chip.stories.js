export default { title: "Skin/Chip/Toggle Discovery Chip" };

export const button = () => `
<button class="chip chip--discovery" type="button" aria-pressed="false">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football</span>
</button>
`;

export const buttonSelected = () => `
<button class="chip chip--discovery" type="button" aria-pressed="true">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football</span>
</button>
`;

export const anchor = () => `
<a class="chip chip--discovery" herf="https://www.ebay.com">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football</span>
</a>
`;

export const anchorSelected = () => `
<a class="chip chip--discovery chip--selected" herf="https://www.ebay.com">
    <img
        class="chip__media"
        src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
        alt=""
    >
    <span class="chip__text">Football <span class="clipped">- Selected</span></span>
</a>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip chip--discovery" type="button" aria-pressed="false">
        <img
            class="chip__media"
            src="https://ir.ebaystatic.com/cr/v/c01/skin/docs/dog_profile_optimized.jpg"
            alt=""
        >
        <span class="chip__text">Football</span>
    </button>
</div>
`;
