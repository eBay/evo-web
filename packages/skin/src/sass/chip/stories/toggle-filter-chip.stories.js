export default { title: "Skin/Chip/Toggle Filter Chip" };

export const button = () => `
<button class="chip chip--filter" type="button" aria-pressed="false">
    <svg class="icon icon--16" width="16" height="16">
        <use href="#icon-sneaker-16"/>
    </svg>
    <span class="chip__text">Football</span>
</button>
`;

export const buttonSelected = () => `
<button class="chip chip--filter" type="button" aria-pressed="true">
    <svg class="icon icon--16" width="16" height="16">
        <use href="#icon-sneaker-16"/>
    </svg>

    <span class="chip__text">Football</span>
</button>
`;

export const anchor = () => `
<a class="chip chip--filter" herf="https://www.ebay.com">
    <svg class="icon icon--16" width="16" height="16">
        <use href="#icon-sneaker-16"/>
    </svg>

    <span class="chip__text">Football</span>
</a>
`;

export const anchorSelected = () => `
<a class="chip chip--filter chip--selected" herf="https://www.ebay.com">
    <svg class="icon icon--16" width="16" height="16">
        <use href="#icon-sneaker-16"/>
    </svg>

    <span class="chip__text">Football <span class="clipped">- Selected</span></span>
</a>
`;

export const RTL = () => `
<div dir="rtl">
    <button class="chip chip--filter" type="button" aria-pressed="false">
        <svg class="icon icon--16" width="16" height="16">
            <use href="#icon-sneaker-16"/>
        </svg>
        <span class="chip__text">Football</span>
    </button>
</div>
`;
