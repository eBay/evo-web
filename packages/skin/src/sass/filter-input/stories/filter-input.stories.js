export default { title: "Skin/Filter Input" };

export const base = () => `
<span class="filter-input" role="search" aria-label="Filter list">
    <span class="textbox textbox--fluid">
        <svg class="icon icon--16" width="16" height="16" aria-hidden="true">
            <use href="#icon-search-16" />
        </svg>
        <input class="textbox__control" placeholder="Filter list" type="search" aria-controls="search-results-container-id" aria-label="Filter list" required/>
        <button
            class="icon-btn icon-btn--transparent icon-btn--small filter-input__clear-btn"
            type="button"
            aria-label="Clear filter"
        >
            <svg aria-hidden="true" class="icon icon--16" width="16" height="16">
                <use href="#icon-clear-16" />
            </svg>
        </button>
    </span>
</span>
`;

export const filled = () => `
<span class="filter-input" role="search" aria-label="Filter list">
    <span class="textbox textbox--fluid">
        <svg class="icon icon--16" width="16" height="16" aria-hidden="true">
            <use href="#icon-search-16" />
        </svg>
        <input class="textbox__control" placeholder="Filter list" type="search" aria-controls="search-results-container-id" aria-label="Filter list" required value="Tenni" />
        <button
            class="icon-btn icon-btn--transparent icon-btn--small filter-input__clear-btn"
            type="button"
            aria-label="Clear filter"
        >
            <svg aria-hidden="true" class="icon icon--16" width="16" height="16">
                <use href="#icon-clear-16" />
            </svg>
        </button>
    </span>
</span>
`;

export const small = () => `
<span class="filter-input filter-input--small" role="search" aria-label="Filter list">
    <span class="textbox textbox--fluid">
        <svg class="icon icon--16" width="16" height="16" aria-hidden="true">
            <use href="#icon-search-16" />
        </svg>
        <input class="textbox__control" placeholder="Filter list" type="search" aria-controls="search-results-container-id" aria-label="Filter list" required />
        <button
            class="icon-btn icon-btn--transparent icon-btn--small filter-input__clear-btn"
            type="button"
            aria-label="Clear filter"
        >
            <svg aria-hidden="true" class="icon icon--16" width="16" height="16">
                <use href="#icon-clear-16" />
            </svg>
        </button>
    </span>
</span>
`;

export const large = () => `
<span class="filter-input filter-input--large" role="search" aria-label="Filter list">
    <span class="textbox textbox--fluid">
        <svg class="icon icon--16" width="16" height="16" aria-hidden="true">
            <use href="#icon-search-16" />
        </svg>
        <input class="textbox__control" placeholder="Filter list" type="search" aria-controls="search-results-container-id" aria-label="Filter list" required />
        <button
            class="icon-btn icon-btn--transparent icon-btn--small filter-input__clear-btn"
            type="button"
            aria-label="Clear filter"
        >
            <svg aria-hidden="true" class="icon icon--16" width="16" height="16">
                <use href="#icon-clear-16" />
            </svg>
        </button>
    </span>
</span>
`;

export const textSpacing = () => `
<span class="filter-input demo-a11y-text-spacing" role="search" aria-label="Filter list">
    <span class="textbox textbox--fluid">
        <svg class="icon icon--16" width="16" height="16" aria-hidden="true">
            <use href="#icon-search-16" />
        </svg>
        <input class="textbox__control" placeholder="Filter list" type="search" aria-controls="search-results-container-id" aria-label="Filter list" required value="Tenni" />
        <button
            class="icon-btn icon-btn--transparent icon-btn--small filter-input__clear-btn"
            type="button"
            aria-label="Clear filter"
        >
            <svg aria-hidden="true" class="icon icon--16" width="16" height="16">
                <use href="#icon-clear-16" />
            </svg>
        </button>
    </span>
</span>
`;

export const RTL = () => `
<div dir="rtl">
    <span class="filter-input" role="search" aria-label="Filter list">
        <span class="textbox textbox--fluid">
            <svg class="icon icon--16" width="16" height="16" aria-hidden="true">
                <use href="#icon-search-16" />
            </svg>
            <input class="textbox__control" placeholder="Filter list" type="search" aria-controls="search-results-container-id" aria-label="Filter list" value="Tenni" required />
            <button
                class="icon-btn icon-btn--transparent icon-btn--small filter-input__clear-btn"
                type="button"
                aria-label="Clear filter"
            >
                <svg aria-hidden="true" class="icon icon--16" width="16" height="16">
                    <use href="#icon-clear-16" />
                </svg>
            </button>
        </span>
    </span>
</div>
`;
