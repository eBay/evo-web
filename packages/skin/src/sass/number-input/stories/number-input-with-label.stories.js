export default { title: "Skin/Number Input/With Label" };

export const base = () => `
<span class="textbox number-input">
    <label for="number-input-1">
        Add a number
    </label>
    <button
        class="icon-btn icon-btn--transparent number-input__decrement"
        tabindex="-1"
        type="button"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-remove-24"/>
        </svg>
    </button>

    <input
        id="number-input-1"
        class="textbox__control"
        type="number"
        min="0"
        value="0"
    >
    <button
        class="icon-btn icon-btn--transparent number-input__increment"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-add-24"/>
        </svg>
    </button>
</span>
`;

export const fluid = () => `
<span class="textbox textbox--fluid number-input">
    <label for="number-input-1">
        Add a number
    </label>
    <button
        class="icon-btn icon-btn--transparent number-input__decrement"
        tabindex="-1"
        type="button"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-remove-24"/>
        </svg>
    </button>

    <input
        id="number-input-1"
        class="textbox__control"
        type="number"
        min="0"
        value="0"
    >
    <button
        class="icon-btn icon-btn--transparent number-input__increment"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-add-24"/>
        </svg>
    </button>
</span>
`;

export const large = () => `
<span class="textbox textbox--large number-input">
    <label for="number-input-1">
        Add a number
    </label>
    <button
        class="icon-btn icon-btn--transparent number-input__decrement"
        tabindex="-1"
        type="button"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-remove-24"/>
        </svg>
    </button>

    <input
        id="number-input-1"
        class="textbox__control"
        type="number"
        min="0"
        value="0"
    >
    <button
        class="icon-btn icon-btn--transparent number-input__increment"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-add-24"/>
        </svg>
    </button>
</span>
`;
