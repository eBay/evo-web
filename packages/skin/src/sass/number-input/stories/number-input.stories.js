export default { title: "Skin/Number Input/Base" };

export const base = () => `
<span class="textbox number-input">
    <button
        class="icon-btn icon-btn--transparent number-input__decrement"
        tabindex="-1"
        type="button"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-remove-24"/>
        </svg>
    </button>

    <input
        aria-label="Number input"
        class="textbox__control"
        type="number"
        min="0"
        value="0"
    >
    <button
        class="icon-btn icon-btn--transparent number-input__increment"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-add-24"/>
        </svg>
    </button>
</span>
`;

export const fluid = () => `
<span class="textbox textbox--fluid number-input">
    <button
        class="icon-btn icon-btn--transparent number-input__decrement"
        tabindex="-1"
        type="button"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-remove-24"/>
        </svg>
    </button>

    <input
        aria-label="Number input"
        class="textbox__control"
        type="number"
        min="0"
        value="0"
    >
    <button
        class="icon-btn icon-btn--transparent number-input__increment"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-add-24"/>
        </svg>
    </button>
</span>
`;

export const large = () => `
<span class="textbox textbox--large number-input">
    <button
        class="icon-btn icon-btn--transparent number-input__decrement"
        tabindex="-1"
        type="button"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-remove-24"/>
        </svg>
    </button>

    <input
        aria-label="Number input"
        class="textbox__control"
        type="number"
        min="0"
        value="0"
    >
    <button
        class="icon-btn icon-btn--transparent number-input__increment"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-add-24"/>
        </svg>
    </button>
</span>
`;

export const withDeleteHidden = () => `
<span class="textbox number-input">
    <button
        class="icon-btn icon-btn--transparent number-input__decrement"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-remove-24"/>
        </svg>
    </button>
    <input
        aria-label="Number input"
        class="textbox__control"
        type="number"
        value="2"
        min="1"
    >
    <button
        class="icon-btn icon-btn--transparent number-input__delete"
        type="button"
        aria-label="Delete item"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <icon-symbol name="delete-24"/>
        </svg>
    </button>

    <button
        class="icon-btn icon-btn--transparent number-input__increment"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-add-24"/>
        </svg>
    </button>
</span>
`;

export const withDeleteShown = () => `
<span class="textbox number-input number-input--show-delete">
    <button
        class="icon-btn icon-btn--transparent number-input__decrement"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-remove-24"/>
        </svg>
    </button>
    <input
        aria-label="Number input"
        class="textbox__control"
        type="number"
        value="1"
        min="1"
    >
    <button
        class="icon-btn icon-btn--transparent number-input__delete"
        type="button"
        aria-label="Delete item"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-delete-24"/>
        </svg>
    </button>

    <button
        class="icon-btn icon-btn--transparent number-input__increment"
        type="button"
        tabindex="-1"
    >
        <svg
            aria-hidden="true"
            class="icon icon--24"
            width="24"
            height="24"
        >
            <use href="#icon-add-24"/>
        </svg>
    </button>
</span>
`;
