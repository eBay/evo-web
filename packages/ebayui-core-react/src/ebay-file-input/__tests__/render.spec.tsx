import React from 'react'
import { composeStories } from '@storybook/react'

import * as stories from './index.stories'
import { render } from '@testing-library/react'

const { Default, WithMockUploads, WithPreviewCards } = composeStories(stories)

describe('<EbayFileInput /> rendering', () => {
    it('renders default story correctly', () => {
        const { container } = render(<Default />)
        expect(container).toMatchSnapshot()
    })

    it('renders with preview cards', () => {
        const { container } = render(<WithPreviewCards />)
        expect(container).toMatchSnapshot()
    })

    it('renders with mock uploads', () => {
        const { container } = render(<WithMockUploads />)
        expect(container).toMatchSnapshot()
    })
})
