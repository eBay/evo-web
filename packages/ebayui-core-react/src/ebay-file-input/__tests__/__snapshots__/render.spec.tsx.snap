// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EbayFileInput /> rendering renders default story correctly 1`] = `
<div>
  <div
    class="file-input"
  >
    <div
      class="file-input__container"
    >
      <div
        class="file-input__upload-icon"
      >
        <svg
          aria-hidden="true"
          class="icon icon--24"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-upload-24"
          />
        </svg>
      </div>
      <div
        class="file-input__content"
      >
        <span
          class="file-input__content-header"
        >
          Upload your files
        </span>
        <span
          class="file-input__content-subheader"
        >
          Supported formats: .jpg, .png
        </span>
        <label
          for="5bee"
        >
          <span
            class="file-input__content-cta"
          >
            <span>
              Click or drag files here to upload
            </span>
          </span>
        </label>
      </div>
    </div>
    <input
      class="file-input__input"
      id="5bee"
      type="file"
    />
  </div>
</div>
`;

exports[`<EbayFileInput /> rendering renders with mock uploads 1`] = `
<div>
  <div
    class="file-input"
  >
    <div
      class="file-input__container"
    >
      <div
        class="file-input__upload-icon"
      >
        <svg
          aria-hidden="true"
          class="icon icon--24"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-upload-24"
          />
        </svg>
      </div>
      <div
        class="file-input__content"
      >
        <span
          class="file-input__content-header"
        >
          <p>
            Multiple files
          </p>
        </span>
        <label
          for="411b"
        >
          <span
            class="file-input__content-cta"
          >
            <span>
              Browse files
            </span>
          </span>
        </label>
      </div>
    </div>
    <input
      class="file-input__input"
      id="411b"
      multiple=""
      type="file"
    />
  </div>
  <div
    class="file-preview-card-group"
  >
    <ul />
  </div>
</div>
`;

exports[`<EbayFileInput /> rendering renders with preview cards 1`] = `
<div>
  <div
    class="file-input"
  >
    <div
      class="file-input__container"
    >
      <div
        class="file-input__upload-icon"
      >
        <svg
          aria-hidden="true"
          class="icon icon--24"
          focusable="false"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-upload-24"
          />
        </svg>
      </div>
      <div
        class="file-input__content"
      >
        <span
          class="file-input__content-header"
        >
          Custom Header
        </span>
        <label
          for="6f62"
        >
          <span
            class="file-input__content-cta"
          >
            <span>
              Click or drag files here to upload
            </span>
          </span>
        </label>
      </div>
    </div>
    <input
      class="file-input__input"
      id="6f62"
      type="file"
    />
  </div>
  <div
    class="file-preview-card-group"
  >
    <ul />
  </div>
</div>
`;
